File: blog_springBoot/src/main/java/com/synergism/blog/core/article/entity/Pagination.java
package com.synergism.blog.core.article.entity;

import lombok.Getter;
import lombok.Setter;

import java.util.List;

/**
 * 分页信息
 */
@Getter
@Setter
public class Pagination {
    //文章信息列表
    private final List<ArticleInformation> articleInformationList;
    //总数
    private final int total;

    public Pagination(List<ArticleInformation> articleInformationList, int total) {
        this.articleInformationList = articleInformationList;
        this.total = total;
    }
}

File: blog_vue/src/model/articles/Pagination.ts
import ArticleInformation from "./ArticleInformation";

export default class Pagination {
  //文章信息列表
  articleInformationList!: Array<ArticleInformation>;
  //总数
  total!: number;

  constructor(
    articleInformationList?: Array<ArticleInformation>,
    total?: number
  ) {
    this.articleInformationList = articleInformationList || [];
    this.total = total || 0;
  }
}
